app/models/manageiq/providers/workflows/automation_manager/workflow.rb

```diff
@@ -10,12 +10,18 @@ def self.create_from_json!(json, **kwargs)
   def run(inputs: {}, userid: "system", zone: nil, role: "automate", object: nil, execution_context: {})
     require "floe"
 
+    manager_ref = SecureRandom.uuid
+
     execution_context = execution_context.dup
+
+    execution_context["Id"]                = manager_ref
```

Comment:
do we want the same name in both places for consistency?
Also, I thought you tended to like an underscore before these entries

Reply:
Execution Id is one of the predefined values in the Context object https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html so I thought it would make sense to map that to our equivalent of a StateMachine Execution (WorkflowInstance)

Reply:
reading the floe code - I really like execution id